### PR TITLE
SystemAPI: owners, timelock upgrades, authorizable functions

### PR DESCRIPTION
**Pull request by felix2feng**

### CODE REVIEW

File: src/api/SystemAPI.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    this.timeLockUpgrade = new TimeLockUpgradeWrapper(web3);
+    this.whitelist = new WhitelistWrapper(web3);
+  }
+
+  /**
+   * Fetches the operational state of Set Protocol. 0 is operational. 1 is shut down.
+   *
+   * @return               Operational State represented as a number
+   */
+  public async getOperationStateAsync(): Promise<BigNumber> {
```
Comment (felix2feng) [inline src/api/SystemAPI.ts]: Start looking here for exposed functions

File: src/wrappers/AuthorizableWrapper.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { Address } from '../types/common';
+
+/**
+ * @title  AuthorizableWrapper
+ * @author Set Protocol
+ *
+ * The AuthorizableWrapper handles all functions and states related to authorizable contracts
+ *
+ */
+export class AuthorizableWrapper {
```
Comment (asoong) [inline src/wrappers/AuthorizableWrapper.ts]: Why do we need a generalized authorizable wrapper? Are you going to add set owner on here as well?

Comment (asoong) [reply src/wrappers/AuthorizableWrapper.ts]: Each of the contracts that extend Authorizable should have their own `getAuthorizedAddresses` address in the abi

File: test/integration/api/OrderAPI.spec.ts [hunk 3]
```diff
@@ -742,7 +742,7 @@ describe('OrderAPI', () => {
       );
 
       const signerWalletAddress = ethUtil.bufferToHex(ethUtil.pubToAddress(signerPublicKey));
-      expect(signerWalletAddress).to.equal(DEFAULT_ACCOUNT);
+      expect(signerWalletAddress).to.equal(DEFAULT_ACCOUNT.toLowerCase());
```
Comment (asoong) [inline test/integration/api/OrderAPI.spec.ts]: why do you have to do this?

Comment (felix2feng) [reply test/integration/api/OrderAPI.spec.ts]: Web3 1.0 returns mixed case addresses, but old tools like ethUtil returns addresses that are lowercase vs mixed case. 

In our default accounts, we previously had lower case addresses, so all of our tests needed to map the accounts to lower case things.

File: src/wrappers/AuthorizableWrapper.ts (hunk 2 above)
Comment (felix2feng) [reply src/wrappers/AuthorizableWrapper.ts]: Hmm... that's true. I'll get rid of it.

**Pull request merged**